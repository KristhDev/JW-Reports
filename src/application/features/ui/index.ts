export { default as uiReducer, UI_INITIAL_STATE, setIsDataExporting, setIsKeyboardVisible, setOldDatetimePicker } from './slice';
export * from './types';