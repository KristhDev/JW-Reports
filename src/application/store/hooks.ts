import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();