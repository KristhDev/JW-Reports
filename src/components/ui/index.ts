export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Title } from './Title';