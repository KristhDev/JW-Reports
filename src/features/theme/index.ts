export { default as themeReducer, setTheme } from './slice';