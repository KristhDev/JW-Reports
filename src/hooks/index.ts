export { default as useAuth } from './useAuth';
export { default as useTheme } from './useTheme';