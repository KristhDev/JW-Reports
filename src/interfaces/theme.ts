export type Theme = 'dark' | 'light' | 'default';

/* Defining the interface for the ThemeState. */
export interface ThemeState {
    colors: Colors;
    deviceTheme: Theme;
    isLoadedTheme: boolean;
    selectedTheme: Theme,
    theme: Theme,
}

/* Defining the interface for the Colors object. */
export interface Colors {
    background: string;
    bottom: string;
    button: string;
    buttonDark: string;
    card: string;
    contentHeader: string;
    focus: string;
    header: string;
    headerText: string;
    icon: string;
    inputText: string;
    linkText: string;
    modal: string;
    modalText: string;
    navbar: string;
    text: string;
    titleSecondary: string;
    titleText: string;
}

/**
 * SetThemePayload is an object with a property called theme that is of type Theme
 * @property {Theme} theme - The theme to set.
 */
export type SetThemePayload = {
    theme: Theme
}