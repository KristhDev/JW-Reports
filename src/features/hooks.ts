import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;