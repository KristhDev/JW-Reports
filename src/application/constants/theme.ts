import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];