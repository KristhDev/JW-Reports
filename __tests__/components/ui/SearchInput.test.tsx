import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react-native';

/* Components */
import { SearchInput } from '../../../src/components/ui';

/* Hooks */
import { useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

const onCleanMock = jest.fn();
const onSearchMock = jest.fn();
const textValue = 'Search test';

/* Mock hooks */
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <SearchInput /> component', () => {
    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors },
        BUTTON_TRANSPARENT_COLOR: 'rgba(255, 255, 255, 0.15)'
    });

    beforeEach(() => {
        render(
            <SearchInput
                onClean={ onCleanMock }
                onSearch={ onSearchMock }
                refreshing={ false }
                searchTerm=""
            />
        );

        jest.clearAllMocks();
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should call onSearch', async () => {

        /* Get text input */
        const input = screen.getByTestId('search-input-text-input');

        await waitFor(() => {

            /* Type a value and submit */
            fireEvent(input, 'onChangeText', textValue);
            fireEvent(input, 'onSubmitEditing');

            /* Check if onSearch is called one time */
            expect(onSearchMock).toBeCalledTimes(1);
        });

        /* Check if onSearch is called with respective value */
        expect(onSearchMock).toBeCalledWith(textValue);
    });

    it('should call onClean when press clear button', async () => {

        /* Get text input and clear touchable */
        const input = screen.getByTestId('search-input-text-input');
        const touchable = screen.getByTestId('search-input-clear-btn');

        await waitFor(() => {

            /* Type value */
            fireEvent(input, 'onChangeText', textValue);
        });

        /* Clear input text when press clear touchable */
        fireEvent.press(touchable);

        /* Check if onClean is called one time */
        expect(onCleanMock).toBeCalledTimes(1);
    });

    it('should change border color when input is focused', async () => {

        /* Get text input and container */
        const input = screen.getByTestId('search-input-text-input');
        const inputContainer = screen.getByTestId('search-input-text-input-container');

        await waitFor(() => {
            fireEvent(input, 'onFocus');

            /* Check if input is focused */
            expect(inputContainer.props.style.borderColor).toBe(darkColors.button);
        });
    });
});