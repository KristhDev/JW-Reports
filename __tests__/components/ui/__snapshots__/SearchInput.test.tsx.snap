// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchInput /> component should to match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 30,
      "paddingHorizontal": 30,
      "width": "100%",
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#BFBFBF",
          "fontSize": 25,
        },
        undefined,
        {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#BFBFBF",
        "flex": 1,
        "flexDirection": "row",
      }
    }
    testID="search-input-text-input-container"
  >
    <TextInput
      autoCorrect={false}
      cursorColor="#C0A7E1"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Buscar"
      placeholderTextColor="#BFBFBF"
      returnKeyType="search"
      style={
        {
          "color": "#FFFFFF",
          "flex": 1,
          "fontSize": 17,
          "paddingLeft": 10,
        }
      }
      testID="search-input-text-input"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      centered={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          false,
          {
            "alignItems": "center",
            "borderRadius": 999,
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          },
        ]
      }
      testID="search-input-clear-btn"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "transparent",
              "fontSize": 30,
            },
            undefined,
            {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
